logdna/resource_archive.go

```diff
@@ -0,0 +1,278 @@
+			Summary:  "Cannot unmarshal response from the remote archive resource",
+			Detail:   err.Error(),
+		})
+		return diags
+	}
+
+	appendError(d.Set("integration", c.Integration), &diags)
+	appendError(d.Set("bucket", c.Bucket), &diags)
+	appendError(d.Set("endpoint", c.Endpoint), &diags)
+	appendError(d.Set("apikey", c.Apikey), &diags)
+	appendError(d.Set("resourceinstanceid", c.Resourceinstanceid), &diags)
+	appendError(d.Set("accountname", c.Accountname), &diags)
+	appendError(d.Set("accountkey", c.Accountkey), &diags)
+	appendError(d.Set("projectid", c.Projectid), &diags)
+	appendError(d.Set("space", c.Space), &diags)
+	appendError(d.Set("accesskey", c.Accesskey), &diags)
+	appendError(d.Set("secretkey", c.Secretkey), &diags)
+	appendError(d.Set("authurl", c.Authurl), &diags)
+	appendError(d.Set("expires", c.Expires), &diags)
+	appendError(d.Set("username", c.Username), &diags)
+	appendError(d.Set("password", c.Password), &diags)
+	appendError(d.Set("tenantname", c.Tenantname), &diags)
+
+	return diags
+}
+
+func resourceArchiveConfigUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	pc := m.(*providerConfig)
+	c := archiveConfig{
+		Integration:        d.Get("integration").(string),
+		Bucket:             d.Get("bucket").(string),
+		Endpoint:           d.Get("endpoint").(string),
+		Apikey:             d.Get("apikey").(string),
+		Resourceinstanceid: d.Get("resourceinstanceid").(string),
+		Accountname:        d.Get("accountname").(string),
+		Accountkey:         d.Get("accountkey").(string),
+		Projectid:          d.Get("projectid").(string),
+		Space:              d.Get("space").(string),
+		Accesskey:          d.Get("accesskey").(string),
+		Secretkey:          d.Get("secretkey").(string),
+		Authurl:            d.Get("authurl").(string),
+		Expires:            d.Get("expires").(int),
+		Username:           d.Get("username").(string),
+		Password:           d.Get("password").(string),
+		Tenantname:         d.Get("tenantname").(string),
+	}
+
+	req := newRequestConfig(
+		pc,
+		"PUT",
+		"/v1/config/archiving",
+		c,
+	)
+
+	_, err := req.MakeRequest()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceArchiveConfigRead(ctx, d, m)
+}
+
+func resourceArchiveConfigDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	pc := m.(*providerConfig)
+	req := newRequestConfig(
+		pc,
+		"DELETE",
+		"/v1/config/archiving",
+		nil,
+	)
+
+	_, err := req.MakeRequest()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func resourceArchiveConfig() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceArchiveConfigCreate,
+		ReadContext:   resourceArchiveConfigRead,
+		UpdateContext: resourceArchiveConfigUpdate,
+		DeleteContext: resourceArchiveConfigDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
```

Comment:
Nit: if we opt for using the schema pattern mentioned in this comment, we could segment the properties here into nested objects for each valid storage `integration`.

Reply:
see other comment